### PR TITLE
Parallel createlistofsize

### PR DESCRIPTION
**Pull request by oviradoi**
This is a pull request for issue #91

### CODE REVIEW

File: tests/FizzWare.NBuilder.Tests/Unit/ParallelListBuilderTests.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using Shouldly;
+using Xunit;
+
+namespace FizzWare.NBuilder.Tests.Unit
+{
+    public class ParallelListBuilderTests
+    {
+        private class MyClass
```
Comment (PureKrome) [inline tests/FizzWare.NBuilder.Tests/Unit/ParallelListBuilderTests.cs]: Yep, it's easy to just create a simple class for this test. But lets try reusing this [existing class called SimpleClass](https://github.com/nbuilder/nbuilder/blob/master/tests/FizzWare.NBuilder.Tests/TestClasses/SimpleClass.cs).

Comment (oviradoi) [reply tests/FizzWare.NBuilder.Tests/Unit/ParallelListBuilderTests.cs]: I have updated the test to use `SimpleClass` instead.

**Comment by mcockrellsana** [comment]
After more investigation, this same issue also seems to occurs in CreateNew, it's not as prevalent, but I've seen issues with int properties being expected to be created with 1 (since you are creating only one instance) but in reality they are either null, 0, or something else.  I think the entire nbuilder package needs a once over to be thread and parallel safe.  @oviradoi would you be willing to fix and write tests for CreateNew as well?

**Comment by oviradoi** [comment]
When creating this pull request, I tried reproducing the same issue for `CreateNew`, but I have not been able to. 

Looking at the code, it looks like `CreateNew` uses the `SetValuesOf<T>(T obj)` method from `SequentialPropertyNamer`. That method always sets the `_sequenceNumber` to 1, so the different threads always use the sequence number of 1.

I can't see this issue happening for `CreateNew`. Maybe, just to be safe, we can use the same fix for `CreateNew`, but I'm not sure how to write a unit test for this case.

**Comment by mcockrellsana** [comment]
I think the instance-based builder is functioning properly in parallelized tests as far as i can tell, it was just that the static implementation had issues referencing the correct instance for auto increment purposes.

**Pull request merged**